server/src/main/java/org/elasticsearch/index/query/VectorGeoShapeQueryProcessor.java

```diff
@@ -86,7 +87,9 @@ protected Query getVectorQueryFromShape(Geometry queryShape, String fieldName, S
 
         @Override
         public Query visit(Circle circle) {
-            throw new QueryShardException(context, "Field [" + fieldName + "] found and unknown shape Circle");
+            // use Point visitor
```

Comment:
geo_shape queries over geo_shape fields do not support circle queries. It should be an error.

Reply:
Sorry, I had misunderstood that, I've rolled back the change